component/http/handler_test.go

```diff
@@ -377,3 +378,102 @@ func Test_extractParamsRawRoute(t *testing.T) {
 	assert.Equal(t, "42", fields["id"])
 	assert.Equal(t, "online", fields["status"])
 }
+
+func Test_getSingleHeaderEncoding(t *testing.T) {
+	testcases := []struct {
```

Comment:
Though I know not all the Patron tests using the table-driven approach are consistent; yet, perhaps we should rather use this approach => https://github.com/beatlabs/patron/blob/c4c83eb1f64e01cf3f4090394597d5acd88a32a1/component/http/handler_test.go#L336
* A map with a meaningful name
* With structs using named fields:
```go
{
	header: json.Type,
	ct: json.TypeCharset,
	dec: json.Decode,
	enc: json.Encode,
	err: nil,
},
```
A bit more verbose but easier to read in my opinion.

Reply:
Which names do you propose for those tests?
Right now the name is directly the header value they are tested against

Reply:
As you prefer.